test/unit/test_quilbase.py

```diff
@@ -266,3 +267,65 @@ def test_center_frequency(self, def_frame: DefFrame, center_frequency: Optional[
         assert def_frame.center_frequency == center_frequency
         def_frame.center_frequency = 432.0
         assert def_frame.center_frequency == 432.0
+
+
+@pytest.mark.parametrize(
+    ("name", "memory_type", "memory_size", "shared_region", "offsets"),
+    [
+        ("ro", "BIT", 1, None, None),
+        ("ro", "OCTET", 5, None, None),
+        ("ro", "INTEGER", 5, "theta", None),
+        ("ro", "BIT", 5, "theta", [(2, "OCTET")]),
+    ],
+    ids=("Defaults", "With-Size", "With-Shared", "With-Offsets"),
+)
+class TestDeclare:
+    @pytest.fixture
+    def declare(
+        self,
+        name: str,
+        memory_type: str,
+        memory_size: int,
+        shared_region: Optional[str],
+        offsets: Optional[Iterable[Tuple[int, str]]],
+    ) -> Declare:
+        return Declare(name, memory_type, memory_size, shared_region, offsets)
+
+    def test_out(self, declare: Declare, snapshot: SnapshotAssertion):
+        assert declare.out() == snapshot
+
+    def test_str(self, declare: Declare, snapshot: SnapshotAssertion):
+        assert str(declare) == snapshot
+
+    def test_asdict(self, declare: Declare, snapshot: SnapshotAssertion):
+        assert declare.asdict() == snapshot
+
+    def test_name(self, declare: Declare, name: str):
+        assert declare.name == name
+        declare.name = "new_name"
+        assert declare.name == "new_name"
+
+    def test_memory_type(self, declare: Declare, memory_type: Optional[str]):
+        assert declare.memory_type == memory_type
+        declare.memory_type = "REAL"
+        assert declare.memory_type == "REAL"
+
+    def test_memory_size(self, declare: Declare, memory_size: Optional[int]):
+        assert declare.memory_size == memory_size
+        declare.memory_size = 100
+        assert declare.memory_size == 100
+
+    def test_shared_region(self, declare: Declare, shared_region: Optional[str]):
+        assert declare.shared_region == shared_region
+        declare.shared_region = "new_shared"
+        assert declare.shared_region == "new_shared"
+
+    def test_offsets(self, declare: Declare, offsets: Optional[Iterable[Tuple[int, str]]]):
+        expected_offsets = [] if offsets is None else offsets
```

Comment:
should be able to use `or`

```suggestion
        expected_offsets = offsets or []
```

Reply:
Done! It's a fair cleanup. I have a phobia of truthy/falsy values in dynamically typed languages, but I need to face my fears.